### PR TITLE
Added monit helper scripts

### PR DESCRIPTION
**Pull request by dwlfrth**
# Description

Adding "monit" helper script to setup PacketFence monit configuration.
By having the scripts here (rather than in Inverse internal documentation), it is easier to update the checks accordingly to the services for each releases.
# Impacts

None... Simply ease monitoring setup
# Delete branch after merge

YES

### CODE REVIEW

**Comment by dwlfrth** [comment]
Review me.
That way, we could start modifying Monit script here rather than internal documentation and avoid having divergences...

**Comment by julsemaan** [comment]
Looks good, not much to review. I'd be happy to press the merge button unless someone thinks this is a bad idea.

File: addons/monit/00_packetfence.sh [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    subject: $SUBJECT_NAME | Monit Alert -- \$EVENT on resource '\$SERVICE'
+    message:
+    Date:        \$DATE
+    Host:        \$HOST
+    Event:       \$EVENT
+    Resource:    \$SERVICE
+    Action:      \$ACTION
+    Description: \$DESCRIPTION
+
+    A copy of this alert have been sent to the PacketFence support team.
```
Comment (fdurand) [inline addons/monit/00_packetfence.sh:36]: I don't want to receive all the monit alert from all the packetfence installation

File: addons/monit/00_packetfence.sh [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    stop program  = "/usr/local/pf/bin/pfcmd service pfmon stop"
+
+check process packetfence-pfqueue with pidfile /usr/local/pf/var/run/pfqueue.pid
+    group PacketFence
+    start program = "/usr/local/pf/bin/pfcmd service pfqueue start" with timeout 60 seconds
+    stop program  = "/usr/local/pf/bin/pfcmd service pfqueue stop"
+
+check process packetfence-radiusd-acct with pidfile /usr/local/pf/var/run/radiusd-acct.pid
+    group PacketFence
+    start program = "/usr/sbin/radiusd -d /usr/local/pf/raddb -n acct" with timeout 60 seconds
```
Comment (fdurand) [inline addons/monit/00_packetfence.sh]: Will not work on debian since radiusd is named freeradius

File: addons/monit/00_packetfence.sh [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    start program = "/usr/sbin/radiusd -d /usr/local/pf/raddb -n auth" with timeout 60 seconds
+    stop program  = "/bin/kill /usr/local/pf/var/run/radiusd.pid"
+    if failed host 127.0.0.1 port 18120 type udp protocol radius
+        secret testing123
+    then alert
+
+check process packetfence-redis_queue with pidfile /usr/local/pf/var/run/redis_queue.pid
+    group PacketFence
+    start program = "/usr/local/pf/bin/pfcmd service redis_queue start" with timeout 60 seconds
+    stop program = "/usr/local/pf/bin/pfcmd service redis_queue stop"
```
Comment (fdurand) [inline addons/monit/00_packetfence.sh:135]: protocol redis is available in monit, so we should probably add it
Something like that:
if failed host 127.0.0.1 port 6380 protocol redis then alert

File: addons/monit/30_packetfence-activeactive.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+cat >> /etc/monit.d/packetfence.monit << EOF
+
+
+# PacketFence active-active clustering checks
+
+check process packetfence-haproxy with pidfile /usr/local/pf/var/run/haproxy.pid
+    group PacketFence
+    start program = "/usr/local/pf/bin/pfcmd service haproxy start" with timeout 60 seconds
+    stop program  = "/usr/local/pf/bin/pfcmd service haproxy stop"
+
```
Comment (fdurand) [inline addons/monit/30_packetfence-activeactive.sh:21]: if failed host 127.0.0.1 port 3306 protocol mysql then alert

File: addons/monit/30_packetfence-activeactive.sh [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+check process packetfence-keepalived with pidfile /usr/local/pf/var/run/keepalived.pid
+    group PacketFence
+    start program = "/usr/local/pf/bin/pfcmd service keepalived start" with timeout 60 seconds
+    stop program  = "/usr/local/pf/bin/pfcmd service keepalived stop"
+
+check process packetfence-radiusd-load_balancer with pidfile /usr/local/pf/var/run/radiusd-load_balancer.pid
+    group PacketFence
+    start program = "/usr/sbin/radiusd -d /usr/local/pf/raddb -n load_balancer" with timeout 60 seconds
```
Comment (fdurand) [inline addons/monit/30_packetfence-activeactive.sh]: debian stuff

File: addons/monit/00_packetfence.sh [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+    stop program  = "/usr/local/pf/bin/pfcmd service pfqueue stop"
+
+check process packetfence-radiusd-acct with pidfile /usr/local/pf/var/run/radiusd-acct.pid
+    group PacketFence
+    start program = "/usr/sbin/radiusd -d /usr/local/pf/raddb -n acct" with timeout 60 seconds
+    stop program  = "/bin/kill /usr/local/pf/var/run/radiusd-acct.pid"
+
+check process packetfence-radiusd with pidfile /usr/local/pf/var/run/radiusd.pid
+    group PacketFence
+    start program = "/usr/sbin/radiusd -d /usr/local/pf/raddb -n auth" with timeout 60 seconds
```
Comment (fdurand) [inline addons/monit/00_packetfence.sh]: debian stuff

File: addons/monit/00_packetfence.sh [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+check process packetfence-config with pidfile /usr/local/pf/var/run/pfconfig.pid
+    group PacketFence
+    start program = "/sbin/service packetfence-config start" with timeout 60 seconds
+    stop program  = "/sbin/service packetfence-config stop"
+
+check process packetfence-redis-cache with pidfile /usr/local/pf/var/run/redis_cache.pid
+    group PacketFence
+    start program = "/sbin/service packetfence-redis-cache start" with timeout 60 seconds
+    stop program  = "/sbin/service packetfence-redis-cache stop"
+
```
Comment (fdurand) [inline addons/monit/00_packetfence.sh:68]: if failed host 127.0.0.1 port 6379 protocol redis then alert

File: addons/monit/00_packetfence.sh (hunk 1 above)
Comment (dwlfrth) [inline addons/monit/00_packetfence.sh:36]: It is a generic message part of the template. Nowhere we're giving out your fdurand@inverse.ca address ;)

File: addons/monit/00_packetfence.sh (hunk 2 above)
Comment (dwlfrth) [inline addons/monit/00_packetfence.sh]: deb who ?

File: addons/monit/00_packetfence.sh (hunk 6 above)
Comment (dwlfrth) [inline addons/monit/00_packetfence.sh]: deb what ?

File: addons/monit/30_packetfence-activeactive.sh (hunk 5 above)
Comment (dwlfrth) [inline addons/monit/30_packetfence-activeactive.sh]: deb why ?

**Comment by dwlfrth** [comment]
Review me

**Comment by julsemaan** [comment]
Looked good to me before, still looks good to me :)

@fdurand can officialize this by merging if it looks good to him too

**Pull request merged**